pkg/api/datasources.go

```diff
@@ -132,3 +132,35 @@ func GetDataSourcePlugins(c *middleware.Context) {
 		c.JSON(200, dsList)
 	}
 }
+
+// Get /api/datasources/name/:name
+func GetDataSourceByName(c *middleware.Context) Response {
+	query := m.GetDataSourceByNameQuery{Name: c.Params(":name"), OrgId: c.OrgId}
+
+	if err := bus.Dispatch(&query); err != nil {
+		if err == m.ErrDataSourceNotFound {
+			return ApiError(404, "Data source not found", nil)
+		}
+		return ApiError(500, "Failed to query datasources", err)
+	}
+
+	ds := query.Result
+
+	return Json(200, &dtos.DataSource{
```

Comment:
The domain model -> dto translation is duplicated twice in this file, can it be refactored out and reused?